worker-document-schema/src/main/resources-yaml/com/microfocus/caf/worker/document/schema/model/schema.yaml

```diff
@@ -0,0 +1,167 @@
+    "$ref": "#/definitions/changeLog"
+additionalProperties: false
+definitions:
+  workerDocument:
+    type: object
+    properties:
+      reference:
+        type: string
+      fields:
+        type: object
+        additionalProperties:
+          type: array
+          items:
+            "$ref": "#/definitions/fieldValue"
+      failures:
+        type: array
+        items:
+          type: object
+          "$ref": "#/definitions/failure"
+      subdocuments:
+        type: array
+        items:
+          type: object
+          "$ref": "#/definitions/workerDocument"
+    additionalProperties: false
+  fieldValue:
+    type: object
+    required:
+    - data
+    properties:
+      data:
+        type: string
+      encoding:
+        type: string
+        default: utf8
+        enum: [utf8, base64, storage_ref, local_ref]
+    additionalProperties: false
+  failure:
+    type: object
+    properties:
+      failureId:
+        type: string
+      failureMessage:
+        type: string
+      failureStack:
+        type: string
+    additionalProperties: false
+  changeLog:
+    type: array
+    items:
+      type: object
+      "$ref": "#/definitions/changeLogEntry"
+  changeLogEntry:
+    type: object
+    properties:
+      name:
+        type: string
+      changes:
+        type: array
+        items:
+          type: object
+          "$ref": "#/definitions/change"
+    additionalProperties: false
+  change:
+    type: object
+    oneOf:
+    - required: [setReference]
+    - required: [addFields]
+    - required: [setFields]
+    - required: [addFailure]
+    - required: [addSubdocument]
+    - required: [insertSubdocument]
+    - required: [updateSubdocument]
+    - required: [removeSubdocument]
+    properties:
+      setReference:
+        type: object
+        properties:
+          value:
+            type: string
+        additionalProperties: false
+      addFields:
+        type: object
+        additionalProperties:
+          type: array
+          items:
+            type: object
+            "$ref": "#/definitions/fieldValue"
+      setFields:
+        type: object
+        additionalProperties:
+          type: array
+          items:
+            type: object
+            "$ref": "#/definitions/fieldValue"
+      removeFields:
+        type: array
+        items:
+          type: string
+      addFailure:
+        type: object
+        "$ref": "#/definitions/failure"
+      setFailures:
+        type: array
+        items:
+          type: object
+          "$ref": "#/definitions/failure"
+      addSubdocument:
+        type: object
+        "$ref": "#/definitions/workerDocument"
+      insertSubdocument:
+        type: object
+        properties:
+          index:
+            type: integer
+          subdocument:
+            type: object
+            "$ref": "#/definitions/workerDocument"
+        additionalProperties: false
+      updateSubdocument:
+        type: object
+        properties:
+          index:
```

Comment:
Mark `index` as mandatory.  Note that in [DocumentWorkerChange.java](https://github.com/CAFDataProcessing/worker-document/blob/v4.1.0/worker-document-shared/src/main/java/com/hpe/caf/worker/document/DocumentWorkerChange.java) it is `int` rather than `Integer` (i.e. it cannot be `null`).

Reply:
index marked as required